src/components/VerticalStepper/VerticalStepper.tsx

```diff
@@ -0,0 +1,213 @@
+import { HTMLAttributes, ReactNode, createContext, useContext } from "react";
+import styled from "styled-components";
+import { Icon } from "@/components";
+
+type StepperType = "numbered" | "bulleted";
+type StepStatus = "active" | "complete" | "incomplete";
+type ContextProps = {
+  type: StepperType;
+};
+
+const StepperContext = createContext<ContextProps>({
+  type: "numbered",
+});
+
+export interface VerticalStepperProps extends HTMLAttributes<HTMLDivElement> {
+  type?: StepperType;
+}
+
+const StepRoot = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  counter-reset: vertical-stepper;
+  width: 100%;
+`;
+
+const VerticalStepper = ({
+  children,
+  type = "numbered",
+  ...props
+}: VerticalStepperProps) => {
+  const value = {
+    type,
+  };
+  return (
+    <StepRoot {...props}>
+      <StepperContext.Provider value={value}>{children}</StepperContext.Provider>
+    </StepRoot>
+  );
+};
+const StepItem = styled.div<{
+  $type: StepperType;
+  $status: StepStatus;
+  $isOpen: boolean;
+}>`
+  position: relative;
+  width: 100%;
+  padding: 0;
+  ${({ theme, $type, $status, $isOpen }) => `
+    padding-bottom: ${
+      theme.click.stepper.vertical[$type].content.space.bottom[
+        $isOpen ? "active" : "default"
+      ]
+    };
+    &:not(:last-of-type) {
```

Comment:
instead of this selector (that is slow) isn't there a way to pass a class to the last one and style that one differently than the rest?

Reply:
Need this as we are not looping through the children